doc/ref/states/writing.rst

```diff
@@ -259,6 +259,49 @@ A State Module must return a dict containing the following keys/values:
 
     States should not return data which cannot be serialized such as frozensets.
 
+Sub State Runs
+--------------
+
+Some states can return multiple state runs from an external engine.
+State modules that extend tools like puppet, chef, ansible, and idem can run multiple external
+states and then return their results individually in the "sub_state_run" portion of their return
+as long as their individual state runs are formatted like salt states with low and high data.
+
+For example, the idem state module can execute multiple idem states
+via it's runtime and report the status of all those runs by attaching them to "sub_state_run" in it's state return.
+These sub_state_runs will be formatted and printed alongside other salt states.
+
+Example:
+
+.. code-block:: python
+
+    state_return = {
+        "result": None,  # The overall status of the external state engine run
+        "comment": None,  # Comments on the overall external state engine run
+        "sub_state_run": {
+            "{external_module}_|-{state_id}_|-{state_name}_|-{external_module_function}": {
+                "result": None,  # The external state run name
+                "comment": None,  # Comment on the external state run
+                "changes": {},  # A dictionary describing the changes made in the external state run
+                "name": "{state_name}",  # Name of the external state run
+                "__id__": "{state_id}",  # ID of the external state run
+                "__run_num__": None,  # run number of the external state run
+            }
```

Comment:
Is there a better way to expect this data to be formatted?  I'm not super familiar with high/low data and analyzing "chunks" from state returns